src/static_array.cr

```diff
@@ -62,12 +62,10 @@ struct StaticArray(T, N)
   # ```
   # StaticArray(Int32, 3).new { |i| i * 2 } # => StaticArray[0, 2, 4]
   # ```
-  def self.new(&block : Int32 -> T)
-    array = uninitialized self
+  def initialize(&block : Int32 -> T)
```

Comment:
See why it's bad to have multiple ways to do the same thing?

Reply:
Not really here, having `&block` is meant for using `block.call`, which is not present here. This could be an error, or easily caught by a linter like ameba.

So here the issue is more: why there is no linter in in crystal-lang/crystal to avoid this things :sweat_smile: .

Anyway, that's out of scope of this PR.

Reply:
My plan was to eventually make it an error if the block is not captured and it has a name.

Meanwhile, it makes no difference. Plus this change has nothing to do with what this PR is solving.